Probable-Futures/data-engineering
# feat: parse and validate mapbox vector tiles

### Description

This PR creates a new package at the root dir called `mapbox-tilesets`. It is responsible on validating Woodwell data against tilesets. There are two methods for validation, and we can use either one:

1- Use [vtQuery](https://github.com/mapbox/vtquery) to query the features at a specific point: (This method is slow since it queries one point at a time).
#### Steps:
```
- Use [tilebelt](https://github.com/mapbox/tilebelt) to get the bounding box of the tileset.
- Parse and read the CSV file (Woodwell data).
- For every point in the CSV file that belongs to the bounding box, call `vtquery` to get the features at this point
- Compare the data in the CSV row with the feature's data obtained from calling `vtquery`

```
2- Fetch tile and read it:
#### Steps:
```
- Fetch a specific tileset
- Read the tileset data as VectorTile.
- Loop through all the features in all the layers in the VectorTile
- Construct a map object that maps each latitude to all points at this latitude
- Calculate the average of all data at a specific latitude.
- Parse and read the CSV file (Woodwell data).
- Loop through all the rows in the CSV file
- Construct a map object that maps each latitude to all points at this latitude
- Calculate the average of all data at a specific latitude
- Compare the data obtained above with the one obtained at step 5
```

### Link to Ticket

[Ticket](https://app.clubhouse.io/probablefutures/story/1825/)

### Steps to QA

- First make sure to fill the required data in `mapbox-tilesets/tests/utils/configs.ts`
- `cd mapbox-tilesets`
- yarn install
- yarn dev

### Screenshot/GIF


https://user-images.githubusercontent.com/18680770/175019741-958fa3cc-4940-4edc-8271-2ee86fa0b8e2.mov
